migration/idempotent/001-base.sql

```diff
@@ -3316,3 +3316,35 @@ END;
 $$
 LANGUAGE PLPGSQL;
 GRANT EXECUTE ON FUNCTION _prom_catalog.insert_metric_row(TEXT, TIMESTAMPTZ[], DOUBLE PRECISION[], BIGINT[]) TO prom_writer;
+
+-- Creates a temporary table (if it doesn't exist) used for ingestion of metrics and traces
+-- Supresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
+-- Temporary table is created using supplied table and schema as prototype.
+-- Returns temporary table name
+CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT)
+    RETURNS TEXT
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+DECLARE
+    temp_table TEXT;               
+BEGIN
+    SET LOCAL log_statement = 'none';
+    -- since table_name gets sanitized by jackc/pgx we need to remove quotes
+    temp_table := FORMAT('%s_temp',REPLACE(table_name, '"', ''));
+    EXECUTE format($sql$CREATE TEMPORARY TABLE IF NOT EXISTS %I (LIKE %I.%I) ON COMMIT DELETE ROWS$sql$,
+                 temp_table, schema_name, table_name);
```

Comment:
1. if `table_name` has double-quotes around it, then they need to be stripped both for the temp_table, AND for the `(LIKE %I.%I)`
2. I'd use `trim(both from table_name, '"')`
3. The biggest issue for me is why the table_name is being passed with double-quotes in the first place. That seems wrong and should be fixed so this issue goes away

Reply:
1. Not really since it's legal to write: `"myschema"."Mytable"`.
The problem only shows up if you try to concat values and one has quotes. `temp_table` becomes `"Mytable"_temp` and that doesn't work.
3. Not sure about that. I think it's a good practice to sanitize user input and that is what PGX Go driver is doing.

Reply:
If your input already has double-quotes, and you use `%I` with format, you'll get double-double-quotes

<img width="335" alt="image" src="https://user-images.githubusercontent.com/4974668/177056372-ac8e9921-0827-42a7-acb2-c1399a704792.png">

Reply:
Which characters is pgx supposed to be escaping? double quotes are valid in a SQL string.

Here, I pass two strings as SQL parameters. One is `Mytable`, and one is `"Mytable"`. Notice that pgx does not wrap `Mytable` in double quotes. In order to end up with double quotes in the SQL string, they had to be passed in that way. 

So, my question is: where in our code are we wrapping the input in double quotes and why?

```
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/jackc/pgx/v5"
)

func main() {
	// urlExample := "postgres://username:password@localhost:5432/database_name"
	conn, err := pgx.Connect(context.Background(), os.Getenv("DATABASE_URL"))
	if err != nil {
		fmt.Fprintf(os.Stderr, "Unable to connect to database: %v\n", err)
		os.Exit(1)
	}
	defer conn.Close(context.Background())

	var one string
	var two string
	err = conn.QueryRow(context.Background(), "select $1, $2", `Mytable`, `"Mytable"`).Scan(&one, &two)
	if err != nil {
		fmt.Fprintf(os.Stderr, "QueryRow failed: %v\n", err)
		os.Exit(1)
	}

	fmt.Printf("one: %s\n", one)
	fmt.Printf("two: %s\n", two)
}
```
<img width="354" alt="image" src="https://user-images.githubusercontent.com/4974668/177057229-ab71f666-0582-40f0-9406-801e1e58d3c6.png">

Reply:
Double quotes should surround a SQL identifier in a statement when mixed-casing should be preserved. This is a different concern from sanitizing user input.

I'm asking that the text input to the function should not be wrapped in double quotes.

Reply:
The code example that you've posted is not really relevant since the problem with quotes is on PG end while interacting with PGX. Here is PGX code that sanitizes argument https://github.com/jackc/pgx/blob/master/internal/sanitize/sanitize.go

Reply:
>I'm asking that the text input to the function should not be wrapped in double quotes.
We are not wrapping any of our inputs into double quotes